main.tf

```diff
@@ -3,6 +3,7 @@
 ##############################################################################
 
 locals {
+  watsonx_assistant_id            = var.existing_watsonx_assistant_instance_crn != null ? data.ibm_resource_instance.existing_assistant_instance[0].id : ibm_resource_instance.watsonx_assistant_instance[0].account_id
```

Comment:
@imprateeksh Isn't this logic going to return the watson assistant instance ID when using an existing instance instead of the account ID?

Reply:
Yes, valid point. The [data source](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/data-sources/resource_instance#attribute-reference) does not supports `account_id` but we get `account_id` when we create a new instance.

@ocofaigh  Do you suggest getting the output only when we create a new instance? Will mention in a comment about data not supporting `account_id `
I think the local variable name should change from `watsonx_assistant_id` to `account_id`

Reply:
Done. Thank you for the suggestion.